File: src/main/java/com/nerdery/jvm/resistance/bots/hopper/PrescriptionData.java
package com.nerdery.jvm.resistance.bots.hopper;

import com.nerdery.jvm.resistance.models.Prescription;

/**
 * A prettified version of the prescription object that simplifies decision making.
 */
class PrescriptionData {
    private boolean prescribedAntibiotics;
    private PatientTemp temp;

    PrescriptionData(boolean prescribedAntibiotics, PatientTemp temp) {
        this.prescribedAntibiotics = prescribedAntibiotics;
        this.temp = temp;
    }

    boolean isPrescribedAntibiotics() {
        return prescribedAntibiotics;
    }

    PatientTemp getTemp() {
        return temp;
    }

    static PrescriptionData fromPrescription(Prescription p) {
        return new PrescriptionData(p.isPrescribedAntibiotics(), PatientTemp.lookupTemp(p.getTemperature()));
    }
}

File: src/main/scala/com/nerdery/jvm/resistance/bots/entrants/hopper/PrescriptionData.scala
package com.nerdery.jvm.resistance.bots.entrants.hopper

import com.nerdery.jvm.resistance.models.Prescription

/**
  * A prettified version of the prescription object that simplifies decision making.
  */
case class PrescriptionData(prescribedAntibiotics: Boolean, temp: PatientTemp)

object PrescriptionData {
  def apply(p: Prescription): PrescriptionData = {
    PrescriptionData(
      prescribedAntibiotics = p.isPrescribedAntibiotics,
      temp = PatientTemp(p.getTemperature)
    )
  }
}
